File: apps/api/app/schemas/device.py
from pydantic import BaseModel, ConfigDict


class DeviceCreate(BaseModel):
    device_name: str
    vendor: str
    os_type: str
    device_type: str
    management_ip: str
    security_domain_id: int
    log_ingest_status: str
    policy_push_capability: bool


class DeviceRead(DeviceCreate):
    model_config = ConfigDict(from_attributes=True)

    id: int

File: packages/contracts/src/device.ts
import { z } from "zod";

export const deviceSchema = z.object({
  id: z.string(),
  deviceName: z.string(),
  vendor: z.literal("Topsec"),
  osType: z.literal("NGTOS"),
  deviceType: z.enum(["edge-fw", "interzone-fw", "waf", "bastion", "vpn", "audit"]),
  managementIp: z.string(),
  securityDomainId: z.string(),
  logIngestStatus: z.enum(["connected", "partial", "disconnected"]).default("connected"),
  policyPushCapability: z.boolean().default(true)
});

export type Device = z.infer<typeof deviceSchema>;
